src/test/java/org/spongepowered/common/regression/SchematicUpgradeTest.java

```diff
@@ -34,23 +34,24 @@
 import org.spongepowered.api.world.schematic.Schematic;
 import org.spongepowered.lwts.runner.LaunchWrapperTestRunner;
 
-import java.io.File;
-import java.io.FileInputStream;
 import java.io.IOException;
+import java.io.InputStream;
 import java.util.zip.GZIPInputStream;
 
 @RunWith(LaunchWrapperTestRunner.class)
 public class SchematicUpgradeTest {
 
     @Test
     public void testUpgradingv1Tov2() throws IOException {
-        File inputFile = new File(this.getClass().getClassLoader().getResource("loadv1.schematic").getFile());
-        DataContainer container = DataFormats.NBT.readFrom(new GZIPInputStream(new FileInputStream(inputFile)));
-        final Schematic v1Schem = DataTranslators.SCHEMATIC.translate(container);
-        inputFile = new File(this.getClass().getClassLoader().getResource("loadv2.schematic").getFile());
-        container = DataFormats.NBT.readFrom(new GZIPInputStream(new FileInputStream(inputFile)));
-        final Schematic v2Schem = DataTranslators.SCHEMATIC.translate(container);
-        assertEquals(v1Schem, v2Schem);
+        final ClassLoader classLoader = this.getClass().getClassLoader();
+        try (final InputStream v1InputStream = classLoader.getResource("loadv1.schematic").openStream();
+             final InputStream v2InputStream = classLoader.getResource("loadv2.schematic").openStream()) {
+                final DataContainer v1Container = DataFormats.NBT.readFrom(new GZIPInputStream(v1InputStream));
```

Comment:
that's better, but `GZIPInputStream` is still not closed

Reply:
:'(
Should be fine now

Reply:
> that's better, but `GZIPInputStream` is still not closed

It gets closed by the `NbtDataFormat.readFrom/writeTo`.

Reply:
@ImMorpheus not at home for 2 weeks, any chance you can just grab the older commit?

Reply:
Nothing wrong with closing already closed stream. It's ok by design.
Also you can just do something like this `new GZIPInputStream(classLoader.getResource("loadv2.schematic").openStream())`, close call shall close all inner streams